automap/src/control_layer/automap_control.rs

```diff
@@ -232,6 +232,10 @@ impl AutomapControlReal {
     ) -> Result<ProtocolInfo<T>, AutomapError> {
         let mut transactors_ref_mut = self.transactors.borrow_mut();
         if let Some(usual_protocol) = self.usual_protocol_opt {
+            debug!(
+                self.logger,
+                "Using a previously discovered protocol: {}", usual_protocol
```

Comment:
This should either be reworded ("Trying a previously-discovered protocol" or something) or it should be moved down to where we know for sure that we really are using a previously-discovered protocol.